ci/azure-travis-template.yaml

```diff
@@ -167,8 +167,8 @@ steps:
       mv *.gcov "$RUN_DIR/"
       # Run codecov
       cd "$RUN_DIR"
-      curl https://codecov.io/bash -o codecov-upload.sh
-      chmod +x codecov-upload.sh
+      curl -Os https://uploader.codecov.io/latest/linux/codecov
+      chmod +x codecov
       # Skip running gcov since we have already done it
-      ./codecov-upload.sh -X gcov
+      ./codecov -X gcov
     displayName: 'Upload coverage information'
```

Comment:
```suggestion
    displayName: 'Upload coverage information'
    env:
      CODECOV_TOKEN: $(CODECOV_TOKEN)
```
That should be enough to make the token accessible to the ``codecov`` script.

Reply:
Commited. Hope this fixes it!

Reply:
Looks like there is an issue with the token `[2021-10-05T16:28:14.801Z] ['error'] There was an error running the uploader: Token found by environment variables with length 16 did not pass validation`

Reply:
That sounds like ``CODECOV_TOKEN`` was not found and the literal string ``$(CODECOV_TOKEN)`` was passed as the token. Just had a look at the code and I can't see why it's not found.

Reply:
Hi @rgommers, could you double check that the new variable was *saved*? (I often forget to click the save button as it's hidden at the very bottom right of the screen when in full-screen.) Thank you.
<img width="486" alt="Screenshot 2021-10-05 at 17 47 41" src="https://user-images.githubusercontent.com/25007832/136067111-b2f8c02d-9110-4c33-a68b-899605f3b652.png">

Reply:
Reloaded the page, and it's there:

<img width="567" alt="image" src="https://user-images.githubusercontent.com/98330/136069554-775fb663-1fe0-4a0e-9a61-4832cbabe964.png">

Reply:
Okay, sorry. Must be something else. Thanks for checking.

Reply:
I am not sure too what's happening. I will try to hunt azure pipelines docs to find something relevant.